backend/climateconnect_api/tests/test_user_views.py

```diff
@@ -48,19 +77,275 @@ def test_signup_api_success(self):
             "last_name": "Tester",
             "country": "Germany",
             "state": "Berlin",
-            "City": "Berlin"
+            "City": "Berlin",
         }
 
-        response = self.client.post(url, data, format='json')
+        response = self.client.post(url, data, format="json")
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
 
     def test_signup_missing_parameter(self):
-        url = reverse('signup-api')
+        url = reverse("signup-api")
         data = {
             "email": "test@testovich.com",
             "password": "testing@2020",
-            "last_name": "Tester"
+            "last_name": "Tester",
         }
 
-        response = self.client.post(url, data, format='json')
+        response = self.client.post(url, data, format="json")
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+
+class TestRecommendedEmail(TestCase):
+    # test the recommended email feature
+    # unit test for generation of project ids, org ids and idea ids
+    # unit test for generation of user ids
+    # test that sends email to MAILJET_ADMIN_EMAIL
+
+    # this function is called once before any test functions are run; the setUp function is called before every test function
+    @classmethod
+    def setUpTestData(cls):
```

Comment:
You don't need to use classmethod here. You can use `def setUp(self)` to setup test data.

Reply:
Your variables will be stored like this: `self.location = LocationFactory()`

Reply:
@ddhanesha I did this, so the setup function would only run once when the class is intitialized. Otherwise the function would be called before every test function and because theres a lot in this function running the tests would probably be quite slow. But not sure how much

Reply:
I think its fine. Usually it doesn't make much difference.